### PR TITLE
Fixing Issues with Sound BitRate Detection, and MediaSource Detection.

### PR DESCRIPTION
**Pull request by ProIcons**

### CODE REVIEW

File: src/main/java/org/tinymediamanager/core/MediaSource.java [hunk 1]
```diff
@@ -47,8 +47,8 @@
   // and our fallback
   UNKNOWN("Unknown");  // @formatter:on
 
-  private static final String START_TOKEN   = "[ .\\-_/\\\\\\[\\(]";
-  private static final String END_TOKEN     = "([ .\\-_/\\\\\\]\\)]|$)";
+  private static final String START_TOKEN   = "[ _\\,\\.\\(\\)\\[\\]\\-]";
```
Comment (mlaggner) [inline src/main/java/org/tinymediamanager/core/MediaSource.java]: a closing bracket ``)]`` makes no sense in START_TOKEN

File: src/main/java/org/tinymediamanager/core/MediaSource.java [hunk 2]
```diff
@@ -47,8 +47,8 @@
...
+  private static final String END_TOKEN     = "([ _\\,\\.\\(\\)\\[\\]\\-]|$)";
```
Comment (mlaggner) [inline src/main/java/org/tinymediamanager/core/MediaSource.java]: an opening bracket ``([`` makes no sonse in END_TOKEN

File: src/main/java/org/tinymediamanager/core/entities/MediaFile.java [hunk 3]
```diff
@@ -1671,12 +1679,24 @@ public void gatherMediaInformation(boolean force) {
           // AAC sometimes codes channels into Channel(s)_Original
           String channels = getMediaInfo(StreamKind.Audio, i, "Channel(s)_Original", "Channel(s)");
           stream.setChannels(StringUtils.isEmpty(channels) ? "" : channels);
+
+          String br = getMediaInfo(StreamKind.Audio, i, "BitRate","BitRate_Maximum","BitRate_Minimum","BitRate_Nominal");
```
Comment (mlaggner) [inline src/main/java/org/tinymediamanager/core/entities/MediaFile.java]: could you move that line just into the try block (not really needed, just to be consistent with the blocks above)

File: src/main/java/org/tinymediamanager/core/entities/MediaFile.java [hunk 4]
```diff
@@ -1759,12 +1779,24 @@ public void gatherMediaInformation(boolean force) {
         stream.setCodec(audioCodec.replaceAll("\\p{Punct}", ""));
         String channels = getMediaInfo(StreamKind.Audio, 0, "Channel(s)");
         stream.setChannels(StringUtils.isEmpty(channels) ? "" : channels + "ch");
+
+        String br = getMediaInfo(StreamKind.Audio, 0, "BitRate","BitRate_Maximum","BitRate_Minimum","BitRate_Nominal");
```
Comment (mlaggner) [inline src/main/java/org/tinymediamanager/core/entities/MediaFile.java]: could you move that line just into the try block (not really needed, just to be consistent with the blocks above)

File: src/main/java/org/tinymediamanager/core/entities/MediaFileAudioStream.java [hunk 5]
```diff
@@ -82,7 +82,7 @@ public int getBitrate() {
   }
 
   public String getBitrateInKbps() {
-    return bitrate + " kbps";
+    return bitrate>0?bitrate + " kbps":"";
```
Comment (mlaggner) [inline src/main/java/org/tinymediamanager/core/entities/MediaFileAudioStream.java]: please use spaces (or our formatter settings) for a more readable code

File: src/main/java/org/tinymediamanager/core/MediaSource.java (hunk 2 above)
Comment (ProIcons) [inline src/main/java/org/tinymediamanager/core/MediaSource.java]: it closes at the end, and it's not my regex, my regex was different i picked it up from kodi. 
I've tested it, and successfully passed all the tests. (i've written some additional tests o v3, please review them)

File: src/main/java/org/tinymediamanager/core/MediaSource.java (hunk 1 above)
Comment (ProIcons) [inline src/main/java/org/tinymediamanager/core/MediaSource.java]: what?

Comment (mlaggner) [inline src/main/java/org/tinymediamanager/core/MediaSource.java]: with your example: Movie.[aXXo]DvDRip.bla.bla.avi this makes sence.
I've just been aware of names like Movie.[DvDRip].bla.bla.avi (where the dvdrip is _inside_ the brackets)

**Comment by mlaggner** [comment]
thanks for the contribution

**Pull request merged**